premigconcor/src/main/java/uk/gov/justice/laadces/premigconcor/Runner.java

```diff
@@ -22,18 +28,38 @@ class Runner implements ApplicationRunner {
     private final MigrationScopeRepository migrationScopeRepository;
     private final ConcorContributionRepository concorContributionRepository;
     private final FdcContributionRepository fdcContributionRepository;
+    private final CaseMigrationRepository caseMigrationRepository;
     private final CsvOutputService csvOutputService;
 
     @Override
-    public void run(ApplicationArguments args) throws Exception {
-        log.info("run() was called");
-        var maatIds = migrationScopeRepository.findAll();
-        log.info("Queried {} maatid from migration database", maatIds.size());
-        var cmsConcor = concorContributionRepository.findLatestIdsByMaatIds(maatIds);
-        log.info("Queried {} concor cases from maat database", cmsConcor.size());
-        var cmsFdc = fdcContributionRepository.findLatestIdsByMaatIds(maatIds);
-        log.info("Queried {} fdc cases from maat database", cmsFdc.size());
-        csvOutputService.writeCaseMigrations("/tmp/premigconcor-ccr.csv", cmsConcor);
-        csvOutputService.writeCaseMigrations("/tmp/premigconcor-fdc.csv", cmsFdc);
+    public void run(final ApplicationArguments args) throws Exception {
+        log.info("Entering run...");
+        final var maatIds = migrationScopeRepository.findAll();
+        log.info("Found {} maatIds from migration database", maatIds.size());
+
+        final var foundConcors = new HashSet<CaseMigration>();
+        final var missingConcors = new TreeSet<Long>();
+        concorContributionRepository.addLatestIdsByMaatIds(maatIds, foundConcors, missingConcors);
+        log.info("Found {} concor cases from maat database, and missing {} maatIds", foundConcors.size(), missingConcors.size());
+        csvOutputService.writeCaseMigrations("/tmp/premigconcor-concorCases.csv", foundConcors);
+        csvOutputService.writeMaatIds("/tmp/premigconcor-concorMissing.csv", MaatId.of(missingConcors));
+
+        final var foundFdcs = new HashSet<CaseMigration>();
+        final var missingFdcs = new TreeSet<Long>();
+        fdcContributionRepository.addLatestIdsByMaatIds(maatIds, foundFdcs, missingFdcs);
+        log.info("Found {} fdc cases from maat database, and missing {} maatIds", foundFdcs.size(), missingFdcs.size());
+        csvOutputService.writeCaseMigrations("/tmp/premigconcor-fdcCases.csv", foundFdcs);
+        csvOutputService.writeMaatIds("/tmp/premigconcor-fdcMissing.csv", MaatId.of(missingFdcs));
```

Comment:
Would it be easier to use if we wrote to the working dir? Or a subfolder of?
Not a big deal, as this is what it is, but to keep things together.

Reply:
Probably. In fact, I'll make them constants at the top of the file, so it's obvious how to change them.